pkg/fleetallocation/controller.go

```diff
@@ -288,3 +292,26 @@ func (c *Controller) allocate(f *stablev1alpha1.Fleet) (*stablev1alpha1.GameServ
 
 	return gs, nil
 }
+
+// patch the labels and annotations of an allocated GameServer with metadata from a FleetAllocation
+func (c *Controller) patchMetadata(gs *stablev1alpha1.GameServer, fam *stablev1alpha1.FleetAllocationMeta) () {
+	// patch ObjectMeta labels
+	if fam.Labels != nil {
+		if gs.ObjectMeta.Labels == nil {
+			gs.ObjectMeta.Labels = make(map[string]string, 1)
+		}
+		for key, value := range fam.Labels {
+			gs.ObjectMeta.Labels[key] = value
+		}
+	}
+
+	// apply annotations patch
+	if fam.Annotations != nil {
+		if gs.ObjectMeta.Annotations == nil {
+			gs.ObjectMeta.Annotations = make(map[string]string, 1)
```

Comment:
@enocom same as above. WDYT?

Reply:
Unless `Annotations` needs to be a small map, the object literal syntax `map[string]string{}` seems more idiomatic to me.

Reply:
We actually don't know the size of the map at all (use configurable).
I'd be inclined to do a `map[string]string{}` myself. 

Personally I only lean on make if I know the size of the map I'm already going to create.

So actually, if we want to keep the `make` we could do:
`gs.ObjectMeta.Annotations = make(map[string]string, len(fam.Annoatations))`
(and similar for the example above)

That would likely actually be the most efficient?

What do you all think?

Reply:
It's a question of readability more than performance in my mind. So if one looks better in the context of Agones, then that's the right one, I think.

Reply:
I had to go back and look - and yep, that's how I did it :smile: 

So I say, let's go with:
`gs.ObjectMeta.Annotations = make(map[string]string, len(fam.Annoatations))`

Since we know the size, and can allocate it upfront, and we won't have to resize down the line.